code/Model/EditableFormField/EditableFileField.php

```diff
@@ -246,7 +246,11 @@ public function getValueFromData()
 
     public function getSubmittedFormField()
```

Comment:
Looks to me that this function definition can be removed because it's defined in the parent class, no?

Reply:
They are different classes — FormFIeld vs FileField

Reply:
Also, the extending class doesn't call the parent method so extending just the parent method would have no effect for the file field.

Reply:
My point is `EditableFileField` extends `EditableFormField`, so this duplicate declaration of `getSubmittedFormField` is not needed at all and can be completely removed from this class.

The parent can then be adjusted to use late static binding (`static::create()`) to replace `$field = SubmittedFormField::create()`.

Reply:
I've renamed the extension point to be unique in each class, more specific in the file class, @dhensby.

I don't think we can use late static binding here. It is true EditableFileField extends EditableFormField, but they each instantiate a different class in their `getSubmittedFormField` method (`SubmittedFormField` vs `SubmittedFileField`) and those classes don't have any connection to the EditableForm/FileField classes using them in terms of inheritance.